numpy/core/src/multiarray/descriptor.c

```diff
@@ -2391,7 +2398,10 @@ arraydescr_reduce(PyArray_Descr *self, PyObject *NPY_UNUSED(args))
         }
         obj = PyUString_FromFormat("%c%d",self->kind, elsize);
     }
-    PyTuple_SET_ITEM(ret, 1, Py_BuildValue("(Nii)", obj, 0, 1));
+    PyTuple_SET_ITEM(ret, 1, Py_BuildValue("(Nii)", obj, 0, !builtin));
```

Comment:
Is it clear that byte order does not apply? What if the pickle is unpikled on a machine with different endianess?